subsys/logging/log_core.c

```diff
@@ -173,14 +187,22 @@ static u32_t timestamp_get(void)
 	return k_cycle_get_32();
 }
 
-int log_init(void)
+int log_init(k_tid_t process_tid)
```

Comment:
We definitely have to separate logger initialization from thread-based processing initialization.
The logger must be be able to log data very early (ideally just after entering C code), while threads could start after system is fully initialized.

Ideally we should end with a call to log_init() in Zephyr startup code and then application might decide how it is going to process gathered data (in the idle, dedicated thread or somewhere else).

Reply:
good point, I've extracted setting thread to external function. log_init() is without parameters and initializes fifo and does not touch thread part. I checked that it will play nicely with #8742